### PR TITLE
Change nerf methods mapping and add thermal predicted image only output

### PR DESCRIPTION
**Pull request by jiaruiyu99**
- change methods mapping in `train_eval_script.py`
- add `thermal` and `thermal_combined` images output

### CODE REVIEW

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/render/renderer.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class RenderedImageModality(Enum):
     depth = "depth"
     accumulation = "accumulation"
     thermal = "thermal"
+    thermal_combined = "thermal_combined"
```
Comment (MalcolmMielle) [inline thermo_nerf/render/renderer.py:22]: Can you make all those uppercase to follow the standard?

Move it to its own file in `thermo_nerf` so you can use it elsewhere

File: thermo_nerf/scripts/train_eval_script.py [hunk 2]
```diff
@@ -25,23 +28,36 @@ class TrainingParameters:
...
     seed: int = 0
     """Seed for the random number generator"""
+    temperature_bounds: list = field(default_factory=lambda: [1.0, 0.0])
+    """Temperature bounds for the dataset"""
+
+    cold: bool = False
+    """Flag to use settings for cold temperatures"""
+
+    def threshold(self):
+        return calculate_threshold(self.data)
```
Comment (MalcolmMielle) [inline thermo_nerf/scripts/train_eval_script.py:40]: Are those used? I don't think `calculate_threshold` exists

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py [hunk 3]
```diff
@@ -348,7 +348,8 @@ def get_image_metrics_and_images(
         )
         combined_thermal = torch.cat([gt_thermal, thermal], dim=1)
 
-        images_dict.update({"thermal": combined_thermal})
+        images_dict.update({"thermal": thermal})
```
Comment (MalcolmMielle) [inline thermo_nerf/thermal_nerf/thermal_nerf_model.py:351]: use your enum `RenderedImageModality` instead of a string here and elsewhere it's used? :)?

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/scripts/renderimagemodality_type.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class RenderedImageModality(Enum):
+    RGB = "rgb"
+    DEPTH = "depth"
+    ACCUMULATION = "accumulation"
+    THERMAL = "thermal"
+    THERMAL_COMBINED = "thermal_combined"
+
+    def __str__(self):
```
Comment (MalcolmMielle) [inline thermo_nerf/scripts/renderimagemodality_type.py:11]: Don't add this function it's not needed. Use `RIM.value` in the code when needed

File: thermo_nerf/scripts/renderimagemodality_type.py
Comment (MalcolmMielle) [inline thermo_nerf/scripts/renderimagemodality_type.py:1]: This file should not be in scripts but directgly in the package: `thermo_nerf`

File: thermo_nerf/scripts/model_type.py
Comment (MalcolmMielle) [inline thermo_nerf/scripts/model_type.py:1]: This file should not be in scripts but directgly in the package: `thermo_nerf`

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py [hunk 5]
```diff
@@ -348,7 +349,10 @@ def get_image_metrics_and_images(
         )
         combined_thermal = torch.cat([gt_thermal, thermal], dim=1)
 
-        images_dict.update({"thermal": combined_thermal})
+        images_dict.update({RenderedImageModality.THERMAL.__str__(): thermal})
```
Comment (MalcolmMielle) [inline thermo_nerf/thermal_nerf/thermal_nerf_model.py:352]: Don't use `__str__` like this directly. Instead in your case, the simpler way is to do

```python
images_dict.update({RenderedImageModality.THERMAL.value: thermal})
images_dict.update(
            {RenderedImageModality.THERMAL_COMBINED.value: combined_thermal}
        )
```

But do you need to use a string?


```python
images_dict.update({RenderedImageModality.THERMAL: thermal})
images_dict.update(
            {RenderedImageModality.THERMAL_COMBINED: combined_thermal}
        )
```
 Should work right?

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py (hunk 5 above)
Comment (jiaruiyu99) [reply thermo_nerf/thermal_nerf/thermal_nerf_model.py]: I get this `Argument of type "dict[RenderedImageModality, Tensor]" cannot be assigned to parameter "m" of type "Iterable[tuple[str, Tensor]]" in function "update"` from pylance and therefore I think we need a string here.

File: thermo_nerf/renderimagemodality_type.py
Comment (MalcolmMielle) [inline thermo_nerf/renderimagemodality_type.py:1]: Why not simply name it like the class? ``render_image_modalities.py`?

File: thermo_nerf/scripts/train_eval_script.py [hunk 6]
```diff
@@ -2,18 +2,25 @@
 from pathlib import Path
 
 import tyro
+
+from thermo_nerf.scripts.model_type import ModelType
+
 from nerfstudio.scripts.train import main
 
 from thermo_nerf.evaluator.evaluator import Evaluator
+from thermo_nerf.model_type import ModelType
```
Comment (MalcolmMielle) [inline thermo_nerf/scripts/train_eval_script.py:11]: You are importing this file multiple time and they are not at the same place. Once if from the remove script one

File: thermo_nerf/model_type.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
```
Comment (MalcolmMielle) [inline thermo_nerf/model_type.py:1]: Why is this file with the evaluator?

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py [hunk 8]
```diff
@@ -33,6 +33,7 @@
 from torchmetrics.image import PeakSignalNoiseRatio
 from torchmetrics.image.lpip import LearnedPerceptualImagePatchSimilarity
 
+from thermo_nerf.renderimagemodality_type import RenderedImageModality
```
Comment (MalcolmMielle) [inline thermo_nerf/thermal_nerf/thermal_nerf_model.py:36]: One place missing on top for the enum

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py [hunk 9]
```diff
@@ -329,7 +330,9 @@ def get_loss_dict(
         thermal_batch = batch["thermal"].to(self.device)
 
         if self.field.pass_thermal_gradients:
-            loss_dict["thermal"] = self.thermal_loss(outputs["thermal"], thermal_batch)
+            loss_dict["thermal"] = self.thermal_loss(
```
Comment (MalcolmMielle) [inline thermo_nerf/thermal_nerf/thermal_nerf_model.py:333]: Why is the loss duct still using a string instead of the enum or at least the string from the enum

File: thermo_nerf/thermal_nerf/thermal_nerf_model.py (hunk 5 above)
Comment (MalcolmMielle) [reply thermo_nerf/thermal_nerf/thermal_nerf_model.py]: It's because the dict typing annotation says the key is a string and now you are forcing an enum... 

Pylance here is not correct ;). But it's ok let's keep a string

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py [hunk 10]
```diff
@@ -269,17 +272,19 @@ def get_image_metrics_and_images(
         metrics_dict = {"psnr": float(psnr.item()), "ssim": float(ssim)}  # type: ignore
         metrics_dict["lpips"] = float(lpips)
 
+        # Follow the same naming convention in nerfstudio, use "img" rather than
```
Comment (MalcolmMielle) [inline thermo_nerf/rgb_concat/concat_nerfacto_model.py:275]: Instead of making a special use case here, change the value of the RGB enum to img:

```
class Modality(Enum):
    RGB = "img"
```

is completely valid :)

File: thermo_nerf/field_heads.py [hunk 11]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class FieldHeadNamesT(Enum):
```
Comment (MalcolmMielle) [inline thermo_nerf/field_heads.py:4]: I wouldn't move this class in this PR:

1. it is outside the scope of the PR, refactoring this adds more work and more modification for little gains ;). You give yourself more work for no reason
2. FieldHeadNamesT is very dependent on the class it was with in the file so it was nice to have that connection this time. When you put it on the base folder, you remove that. At the very least it should be in the model folder.

All in all I would revert this change

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py (hunk 10 above)
Comment (jiaruiyu99) [reply thermo_nerf/rgb_concat/concat_nerfacto_model.py:275]: For concat-nerf, we overwrite the get_output() function of nerfacto model in nerfstudio. 
In concat-nerf model, we have to create and  use `output` and `image_dict`.

The naming convention for `output` in nerfstudio is output["rgb"].
The naming convention for `image_dict` in nerfstudio is image_dict["img"].

In this way, we cannot just set Modality.RGB.value as "img", as we are mainly using Modality.RGB.value for `output`.

Therefore, I guess the easiest way is to just use a special case here.

File: thermo_nerf/evaluator/evaluator.py [hunk 12]
```diff
@@ -77,7 +77,7 @@ def _compute_metrics(
             ) = self._pipeline.model.get_image_metrics_and_images(outputs, batch)
 
             # Save imgs
-            images_dict["rgb"] = images_dict.pop("img")
+            images_dict[RenderedImageModality.RGB.value] = images_dict.pop("img")
```
Comment (jiaruiyu99) [inline thermo_nerf/evaluator/evaluator.py:80]: Here is the code to change the naming convention for `image_dict["img"]` in nerfstudio to our own modality types. This is actually for thermonerf and nerfacto model as they directly use the nerfacto model in nerfstudio. 

For concat-nerf, if without this code in the evaluator, we could directly use `image_dict[modality.RGB.value]`. With this code, we should use  `image_dict["img"]` in the conat-nerfacto-model.

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py [hunk 13]
```diff
@@ -221,7 +222,7 @@ def get_metrics_dict(self, outputs, batch):
         metrics_dict = {}
         gt_rgb = batch["image"].to(self.device)  # RGB or RGBA image
 
-        predicted_rgb = outputs["rgb"]
+        predicted_rgb = outputs[RenderedImageModality.RGB.value]
```
Comment (jiaruiyu99) [inline thermo_nerf/rgb_concat/concat_nerfacto_model.py:225]: To have the predicted rgb images, we could only use `output["rgb"]` and not `output["img"]`. Besides, we could not change it to `output["img"]` as it is created in nerfstudio.

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py (hunk 10 above)
Comment (MalcolmMielle) [reply thermo_nerf/rgb_concat/concat_nerfacto_model.py:275]: I see now, that's annoying. Ok for now then :/

File: thermo_nerf/evaluator/evaluator.py (hunk 12 above)
Comment (MalcolmMielle) [reply thermo_nerf/evaluator/evaluator.py:80]: Then we should switch to "img" instead of "rgb" so we avoid those pop operation which are a little unclear and instead we put your "special case" just for rgb in the output of concat-nerf? Does that sounds good. This way it's closer to nerfstudio but also it's less places with switch between enum values

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py (hunk 10 above)
Comment (MalcolmMielle) [inline thermo_nerf/rgb_concat/concat_nerfacto_model.py:275]: Remove unnecessary comments

File: thermo_nerf/rgb_concat/concat_nerfacto_model.py [hunk 14]
```diff
@@ -269,17 +272,19 @@ def get_image_metrics_and_images(
         metrics_dict = {"psnr": float(psnr.item()), "ssim": float(ssim)}  # type: ignore
         metrics_dict["lpips"] = float(lpips)
 
+        # Follow the same naming convention in nerfstudio, use "img" rather than
+        # RenderedImageModality.RGB.value
         images_dict = {
             "img": combined_rgb,
```
Comment (MalcolmMielle) [inline thermo_nerf/rgb_concat/concat_nerfacto_model.py:278]: Use the enum.

**Review by MalcolmMielle — CHANGES_REQUESTED** [review]

**Review by MalcolmMielle — APPROVED** [review]
